src/Compilers/Core/Portable/Symbols/WellKnownMemberNames.cs

```diff
@@ -398,5 +398,9 @@ public static class WellKnownMemberNames
         /// The name of a type synthesized for a top-level statements entry point method.
         /// </summary>
         public const string TopLevelStatementsEntryPointTypeName = "Program";
+
+        internal const string LockTypeName = "Lock";
+        internal const string EnterLockScopeMethodName = "EnterLockScope";
+        internal const string LockScopeTypeName = "Scope";
```

Comment:
We'll probably want to see if API review wants these to be public.

Reply:
> We'll probably want to see if API review wants these to be public.

We can certainly consider that, but we don't have to make them public "by default".